youtube_dl/utils.py

```diff
@@ -1835,6 +1836,12 @@ def write_json_file(obj, fn):
                 os.unlink(fn)
             except OSError:
                 pass
+        os.chmod(tf.name,
+                 stat.S_IRUSR |
+                 stat.S_IWUSR |
+                 stat.S_IRGRP |
+                 stat.S_IROTH)
+
```

Comment:
1. No umask respected.
2. There is no `os.chmod` in python 3.2 according to python [docs](https://docs.python.org/3/library/os.html#os.chmod).
3. flake8.

Reply:
umask should be good now.
flake8 no longer complaines (sorry about that one!)
Not sure what you mean about os.chmod not being in python3.2?
The link takes me to 3.8 and the python 3.2 docs [clearly show its there](https://docs.python.org/3.2/library/os.html)
If I remember when I last checked tf.name returned a string object so that should work.